lbrynet/core/BlobManager.py

```diff
@@ -143,9 +144,11 @@ def set_length(length):
 
     def blob_completed(self, blob, next_announce_time=None):
         if next_announce_time is None:
-            next_announce_time = time.time()
-        return self._add_completed_blob(blob.blob_hash, blob.length,
-                                        time.time(), next_announce_time)
+            next_announce_time = time.time() + self.hash_reannounce_time
+        d = self._add_completed_blob(blob.blob_hash, blob.length,
+                                     time.time(), next_announce_time)
+        d.addCallback(lambda _: self.hash_announcer.immediate_announce([blob.blob_hash]))
```

Comment:
I set next_announce_time to be the hour in a future so that avoids the run_manage_loop one.